src/QueryBuilder.php

```diff
@@ -29,6 +29,10 @@ class QueryBuilder extends Builder
      */
     public function __construct($builder, ?Request $request = null)
     {
+        if ($builder instanceof \Illuminate\Database\Eloquent\Relations\Relation) {
```

Comment:
You could add a comment to explain why this conditional exists. Without the context of this PR the repeated call to `$builder->getQuery()` could appear a little bit confusing.